rules/aip0158/response_first_field_must_be_plural.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0158
+
+import (
+	"fmt"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var responseFirstFieldMustBePlural = &lint.MessageRule{
+	Name:   lint.NewRuleName(158, "response-first-field-must-be-plural"),
+	OnlyIf: isPaginatedResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// Rule check: for all messages that end in Response and contain a next_page_token field.
+		// Throw a linter warning if, the first field in the message is not named according to plural(message_name.to_snake().split('_')[1:-1]).
+		nextPageToken := m.FindFieldByName("next_page_token")
+		if nextPageToken == nil {
+			return nil
+		}
+
+		firstField := m.FindFieldByNumber(1)
+		pluralize := pluralize.NewClient()
+		if !pluralize.IsPlural(firstField.GetName()) {
```

Comment:
This may give unhelpful feedback in some situations.
For example, let's say I did...

```proto
message ListBooksResponse {
  string next_page_token = 1;
  repeated Book books = 2;
}
```

The thing the linter should tell me to do is to reverse the order of the fields. What it will actually do is still me to rename the first field to `string next_page_tokens`.

**Note:** This is probably a situation where there are several rules and at most one of them should complain to avoid giving users misleading output.

Reply:
I still think there are some cases where this linter rule should be quiet (and defer to others to be loud).

I think we should probably only grouse about the name if the field is a repeated non-primitive, and a different rule will complain about those errors.